### PR TITLE
Fix some typos and add links in intro_to_datalad.ipynb

### PR DESCRIPTION
**Pull request by rly**
While I was going through the intro_to_datalad tutorial (linked to from [datalad.org](https://www.datalad.org/)), I found a few typos and references to sections that did not exist in the notebook. I fixed the typos and added links to the DataLad Handbook for the referenced sections.

Thanks for making this notebook!

### CODE REVIEW

**Comment by rly** [comment]
I see that some of the missing links are added in #21. My bad - didn't see that.

File: notebooks/intro_to_datalad.ipynb [hunk 1]
```diff
@@ -1171,8 +1171,8 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "Read up more about the YODA principles and the `yoda` configuration in the `yoda`\n",
-    "section.\n",
+    "Read up more about the YODA principles and the `yoda` configuration in the \n",
+    "[`yoda` section of the DataLad Handbook](https://handbook.datalad.org/en/latest/basics/101-127-yoda.html).\n",
```
Comment (yarikoptic) [inline notebooks/intro_to_datalad.ipynb]: we have some trickery to centralize such "shortcuts" for the cases when sections would get renumbered etc
```suggestion
    "[`yoda` section of the DataLad Handbook](https://handbook.datalad.org/en/latest/r.html?yoda).\n",
```

File: notebooks/intro_to_datalad.ipynb [hunk 2]
```diff
@@ -1196,7 +1196,7 @@
    "source": [
     "The last piece is the code to run on the data and produce results. For this, here is a\n",
     "k-means classification analysis script written in Python. You can find this analysis\n",
-    "in more detail in the `yoda_project` section:"
+    "in more detail in the [`yoda_project` section of the DataLad Handbook](https://handbook.datalad.org/en/latest/basics/101-130-yodaproject.html):"
```
Comment (yarikoptic) [inline notebooks/intro_to_datalad.ipynb:1199]: but not this one ;)

**Comment by yarikoptic** [comment]
Thank you @rly! What a shame for the "codespell advocate" (I refer to myself here) ;)

**Pull request merged**